omnigibson/transition_rules.py

```diff
@@ -369,213 +517,192 @@ def transition(self, individual_objects, group_objects):
-        group_filters = {category: CategoryFilter(category) for category in obj_requirements.keys()}
+        @classproperty
+        def cp_output_system(cls):
+            return output_system
+        kwargs["output_system"] = cp_output_system
+
+        # Override the particle requirements if specified
+        if particle_requirements is not None:
+            @classproperty
+            def cp_particle_requirements(cls):
+                return particle_requirements
+            kwargs["particle_requirements"] = cp_particle_requirements
+
+        # Override the object requirements and group filters class properties based on the requested object
+        # requirements if requested
+        if object_requirements is not None:
+            @classproperty
+            def cp_object_requirements(cls):
+                return object_requirements
+            kwargs["object_requirements"] = cp_object_requirements
+
+            @classproperty
+            def cp_group_filters(cls):
+                return {category: CategoryFilter(category) for category in object_requirements.keys()}
+            kwargs["group_filters"] = cp_group_filters
+
+        # Create and return the class
+        return subclass_factory(name=name, base_classes=cls, **kwargs)
+
+    @classproperty
+    def output_system(cls):
+        """
+        Returns:
+            str: Corresponding output system for this specific blender rule
+        """
+        raise NotImplementedError()
 
-        # Store internal variables
-        self.output_system = output_system
-        self.particle_requirements = particle_requirements
-        self.obj_requirements = obj_requirements
+    @classproperty
+    def particle_requirements(cls):
+        """
+        Returns:
+            dict: Maps physical particle system name to the minimum number of physical particles required in order to
+                successfully blend
+        """
+        # Default is empty dictionary
+        return dict()
 
-        # Store a cached dictionary to check blender volumes so we don't have to do this later
-        self._check_in_volume = dict()
+    @classproperty
+    def object_requirements(cls):
+        """
+        Returns:
+            dict: Maps object category names to minimum number of that type of object required in order to
+                successfully blend
+        """
+        # Default is empty dictionary
+        return dict()
 
-        # Call super method
-        super().__init__(individual_filters=individual_filters, group_filters=group_filters)
+    @classproperty
+    def individual_filters(cls):
+        # We want to filter for any object that is included in @obj_requirements and also separately for blender
+        return {"blender": CategoryFilter("blender")}
 
-    def condition(self, individual_objects, group_objects):
+    @classmethod
+    def condition(cls, individual_objects, group_objects):
         # TODO: Check blender if both toggled on and lid is closed!
```

Comment:
I think we should check if the blender is toggledOn here. Otherwise I am not sure what is the trigger event? 

I am okay with leaving the "lid is closed" requirement out because some kitchen appliance might not have a lid.

Reply:
Good point! Added.

Reply:
Sort of related: we need to check it's ToggledOn is True in the current timestep AND False in the previous timestep. 
For magic stirring rod rule, it's Touching.

Reply:
done